js/main.js

```diff
@@ -239,10 +265,177 @@ if ($aTagCheckmark) {
+      };
+      allPotentialPlaylists.push(potentialPlaylistObject);
+    }
+  }
+
+  if (workoutMode === 'hardcore') {
+    thirdLength = Math.floor(trackList.length / 3);
+    var topThirdList = [];
+    for (var o = trackList.length - thirdLength; o < trackList.length; o++) {
+      topThirdList.push(trackList[o]);
+    }
+    numSongs = numSections;
+
+    allPotentialPlaylists = [];
+    for (var p = 0; p < 100; p++) {
+      var topThirdListShuffled = shuffle(topThirdList);
+      potentialPlaylistTracks = topThirdListShuffled.slice(0, numSongs);
+
+      potentialPlaylistTotalDuration = 0;
+      for (var q = 0; q < potentialPlaylistTracks.length; q++) {
+        potentialPlaylistTotalDuration += potentialPlaylistTracks[q].duration_ms;
+      }
+      playlistDurationDeviation = Math.abs(potentialPlaylistTotalDuration - duration);
+      potentialPlaylistObject = {
+        tracks: potentialPlaylistTracks,
+        totalDuration: potentialPlaylistTotalDuration / 60000,
+        deviation: playlistDurationDeviation / 60000
+      };
+      allPotentialPlaylists.push(potentialPlaylistObject);
+    }
+
+  }
+
+  if (workoutMode === 'chill') {
+    thirdLength = Math.floor(trackList.length / 3);
+    var bottomThirdList = [];
+    for (var t = 0; t < thirdLength; t++) {
+      bottomThirdList.push(trackList[t]);
+    }
+    numSongs = numSections;
+
+    allPotentialPlaylists = [];
+    for (var r = 0; r < 100; r++) {
+      var bottomThirdListShuffled = shuffle(bottomThirdList);
+      potentialPlaylistTracks = bottomThirdListShuffled.slice(0, numSongs);
+
+      potentialPlaylistTotalDuration = 0;
+      for (var s = 0; s < potentialPlaylistTracks.length; s++) {
+        potentialPlaylistTotalDuration += potentialPlaylistTracks[s].duration_ms;
+      }
+      playlistDurationDeviation = Math.abs(potentialPlaylistTotalDuration - duration);
+      potentialPlaylistObject = {
+        tracks: potentialPlaylistTracks,
+        totalDuration: potentialPlaylistTotalDuration / 60000,
+        deviation: playlistDurationDeviation / 60000
+      };
+      allPotentialPlaylists.push(potentialPlaylistObject);
+    }
+
+  }
+
+  allPotentialPlaylists.sort((a, b) => (a.deviation - b.deviation));
+  var chosenPlaylistInfo = allPotentialPlaylists[0].tracks;
+  var chosenPlaylistIDs = [];
+
+  for (var n = 0; n < chosenPlaylistInfo.length; n++) {
+    chosenPlaylistIDs.push(chosenPlaylistInfo[n].id);
+  }
+
+  if (workoutMode === 'ramp-up-n-down') {
+    var list1 = [];
+    var list2 = [];
+    for (var u = 0; u < chosenPlaylistIDs.length; u++) {
+      if (u % 2 === 0) {
+        list1.push(chosenPlaylistIDs[u]);
+      } else {
+        list2.push(chosenPlaylistIDs[u]);
+      }
+    }
+    list2.reverse();
+    chosenPlaylistIDs = list1.concat(list2);
+  }
+
+  data.playlistTrackIDs = chosenPlaylistIDs;
+  createFinalTrackIDsURLString(data.playlistTrackIDs);
+  getTracks(data.finalTrackIDsURLString);
+}
```

Comment:
It feels like there probably some re-usable logic for the different `workoutMode`s, so I'd suggest taking another look at ways to abstract some of the work into a re-useable function.